### PR TITLE
Remove Unrestored/UnrestoredTypeKey flags

### PR DESCRIPTION
**Pull request by elinor-fung**
These should not be needed now that we NGEN support is gone.

### CODE REVIEW

File: src/coreclr/vm/clsload.cpp [hunk 1]
```diff
@@ -2059,7 +2058,6 @@ TypeHandle ClassLoader::LoadTypeDefOrRefThrowing(ModuleBase *pModule,
         PRECONDITION(level > CLASS_LOAD_BEGIN && level <= CLASS_LOADED);
 
         POSTCONDITION(CheckPointer(RETVAL, NameHandle::OKToLoad(typeDefOrRef, tokenNotToLoad) && (fNotFoundAction == ThrowIfNotFound) ? NULL_NOT_OK : NULL_OK));
-        POSTCONDITION(level <= CLASS_LOAD_UNRESTORED || RETVAL.IsNull() || RETVAL.IsRestored());
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/vm/clsload.cpp:2062]: Should we still permit `RETVAL.IsNull()`?

Comment (elinor-fung) [reply src/coreclr/vm/clsload.cpp:2062]: Permitting it or not it should be handled by the `CheckPointer` above - or do you mean something else?

Comment (AaronRobinsonMSFT) [reply src/coreclr/vm/clsload.cpp:2062]: Oh right. Good point.

File: src/coreclr/vm/methoditer.cpp [hunk 2]
```diff
@@ -108,9 +108,6 @@ BOOL LoadedMethodDescIterator::Next(
```
Comment (davidwrighton) [inline src/coreclr/vm/methoditer.cpp:108]: Remove the comment at line 100 around being really careful in the presence of unrestored types.

File: src/coreclr/debug/daccess/dacdbiimpl.cpp [hunk 3]
```diff
@@ -1490,16 +1490,11 @@ void DacDbiInterfaceImpl::GetTypeHandles(VMPTR_TypeHandle  vmThExact,
     // If we can't find the class, return the proper HR to the right side. Note: if the class is not a value class and
     // the class is also not restored, then we must pretend that the class is still not loaded. We are gonna let
     // unrestored value classes slide, though, and special case access to the class's parent below.
-    if ((pThApprox->IsNull()) || ((!pThApprox->IsValueType()) && (!pThApprox->IsRestored())))
+    if (pThApprox->IsNull())
```
Comment (davidwrighton) [inline src/coreclr/debug/daccess/dacdbiimpl.cpp:1493]: Adjust comment above. Remove the "Note:"

**Review by davidwrighton — APPROVED** [review]
Looks good to me. I've noticed a few comments that need updating,  (and there are doubtless others) but this seems good enough to checkin.

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Pull request merged**